Search for Item with Type Text in All Subfolders

NatashaDudi opened the pull request:
Changing the method from looking at direct children to looking at all children until one is found that has an item of type "Text". Making sure that if a instance has no text item, there is an error message even if the instance has children in subfolders.

Commit:
Search for Item with Type Text in All Subfolders

Changing the method from looking at direct children to looking at all children until one is found that has an item of type "Text". Making sure that if a instance has no text item, there is an error message.

Landschaft commented on src/main/code.ts:
```diff
@@ -20,23 +20,26 @@ if (figma.currentPage.selection.length != 1) {
 	  var textNode = getFirstChildOfTypeText(figma.currentPage.selection[0])
-	  if (textNode === undefined) {
+	  if (textNode === undefined || textNode == null) {
 		figma.closePlugin("🛑 Selected node has to child of type TEXT");
```
Ah, here is the typo I was referring to, the other day. Can you please fix this as well?

Landschaft replied:
```suggestion
		figma.closePlugin("🛑 Selected node has to have a child of type TEXT");
```

Landschaft commented on src/main/code.ts:
```diff
@@ -116,7 +119,7 @@ if (figma.currentPage.selection.length != 1) {
   
   function getFirstChildOfTypeText(sceneNode: SceneNode) {
 	if ("children" in sceneNode) {
```
Since property `children` is not used any more, this check is no longer necessary. 

Moreover, I think we can now inline the call to method `getFirstChildOfTypeText`.

NatashaDudi replied:
The function "findOne()" can only be used when the scene node has children. Therefore the if-else-statement is still necessary.

Landschaft replied:
They don't say that in the [docs](https://www.figma.com/plugin-docs/api/properties/nodes-findone/) – you probably found this by experiment?

Landschaft commented:
Don't forget to adjust the how-to section in the README

NatashaDudi committed:
Update src/main/code.ts

update error messages

Co-authored-by: Landschaft <fabian.nart@gmail.com>

Commit:
update README

Since the text label does not have to be a direct child anymore the text in brackets can be deleted.

Commit:
Merge branch 'Text-in-Frames' of https://github.com/ergon/figma-table-builder into Text-in-Frames

NatashaDudi committed:
Delete code.js

Landschaft commented on src/main/code.ts:
```diff
@@ -116,7 +119,7 @@ if (figma.currentPage.selection.length != 1) {
-	  return sceneNode.children.find(node => node.type === "TEXT") as TextNode
+	  return sceneNode.findOne(node => node.type === "TEXT") as TextNode
 	} else {
 	  return undefined
```
To simplify check in calling code

```suggestion
	  return null
```

NatashaDudi committed:
Update src/main/code.ts

Co-authored-by: Landschaft <fabian.nart@gmail.com>

Commit:
update README, only check for null
